src/v/rpc/connection_cache.h

```diff
@@ -102,11 +102,22 @@ inline ss::shard_id connection_cache::shard_for(
   ss::shard_id src_shard,
   model::node_id n,
   ss::shard_id total_shards) {
-    static const constexpr size_t vnodes = 3;
+    if (ss::smp::count <= 8) {
```

Comment:
sorry if this question was already asked; i think this pr has been up long enough i may have forgotten the answer:

before we always hashed, why do we only now hash when there are more than 8 cores? even with a small number of cores don't we still want to do some probabilistic spreading?

Reply:
For lower number of cores we just do connection per core

Reply:
is it not still valuable to still hash for small number of cores in case `src_shard` isn't well distributed? i think that's what my question was intended to ask.

Reply:
the idea is that when there is a connection per shard the best way is to use core local connection to avoid cross core communication